packages/next/client/link.tsx

```diff
@@ -282,6 +282,11 @@ function Link(props: React.PropsWithChildren<LinkProps>) {
   } = {
     ref: setRef,
     onClick: (e: React.MouseEvent) => {
+      if (!e) {
+        if (process.env.NODE_ENV !== 'production') {
```

Comment:
I think these `if` statements should be swapped so that the whole thing can be removed in production